src/scripts/scripts/zone/hellfire_citadel/hellfire_ramparts/boss_watchkeeper_gargolmar.cpp

```diff
@@ -126,20 +151,29 @@ struct boss_watchkeeper_gargolmarAI : public ScriptedAI
         if (MortalWound_Timer < diff)
         {
             DoCast(me->getVictim(),HeroicMode ? H_SPELL_MORTAL_WOUND : SPELL_MORTAL_WOUND);
-            MortalWound_Timer = 5000+rand()%8000;
+            MortalWound_Timer = urand(5000, 13000);
         }else MortalWound_Timer -= diff;
 
         if (Surge_Timer < diff)
         {
             DoScriptText(SAY_SURGE, me);
 
-            if (Unit* target = SelectUnit(SELECT_TARGET_RANDOM,0,GetSpellMaxRange(SPELL_SURGE), true))
-                DoCast(target,SPELL_SURGE);
-
-            Surge_Timer = 5000+rand()%8000;
+            if (Unit * target = SelectUnit(SELECT_TARGET_FARTHEST, 0, 40.0f, true, 0, 1.0f))
+            {
```

Comment:
Try adding `m_creature->GetMotionMaster()->MovementExpired();` to fix charge animation